unet-instance-segmentation/experiment/trainer.py

```diff
@@ -87,15 +101,92 @@ def __init__(self):
         self.min_contour_points = 20
 
     def prepare_data_for_training(self):
-        self._download_data()
+        self.segmentation_dataset = self.get_segmentation_dataset()
+        if self.segmentation_dataset is not None:
+            self.download_segmentation_data_into_masks()
+        else:
+            self.mask_dataset, self.image_dataset = self.get_mask_image_datasets()
+            if self.mask_dataset is not None and self.image_dataset is not None:
+                self._download_mask_image_datasets()
+                self._get_mask_image_filenames()
+
+        self.classes = get_classes(
+            self.experiment, has_segmentation_dataset=self.has_segmentation_dataset
```

Comment:
`has_segmentation_dataset` is now always to None from the __init__ fn

Reply:
its value is modified in get_segmentation_dataset()